sky/serve/replica_managers.py

```
@@ -59,6 +60,7 @@ def launch_cluster(replica_id: int,
                    task_yaml_path: str,
                    cluster_name: str,
                    resources_override: Optional[Dict[str, Any]] = None,
+                   retry_until_up: bool = True,
```

Why is `retry_until_up` part of the spot PR?

---

Previously we always set `retry_until_up=True`. And if the `retry_until_up` is set to `True`, it will keep retrying a region without spot capacity forever. For SpotHedge, we want it to fail fast after 1 trial and quickly mark it as preemptive.

---

oh yeah yes.